recipes/freetype/all/conanfile.py

```diff
@@ -71,27 +76,49 @@ def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         os.rename("{0}-{1}".format(self.name, self.version), self._source_subfolder)
 
+    def _patch_sources(self):
+        # Do not accidentally enable dependencies we have disabled
+        cmakelists = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        find_harfbuzz = "find_package(HarfBuzz {})".format("1.3.0" if tools.Version(self.version) < "2.10.2" else "${HARFBUZZ_MIN_VERSION}")
+        tools.replace_in_file(cmakelists, find_harfbuzz, "")
+        tools.replace_in_file(cmakelists, "if (HARFBUZZ_FOUND)", "if(0)")
+        if not self.options.with_png:
+            tools.replace_in_file(cmakelists, "find_package(PNG)", "")
+            tools.replace_in_file(cmakelists, "if (PNG_FOUND)", "if(0)")
+        if not self.options.with_zlib:
+            tools.replace_in_file(cmakelists, "find_package(ZLIB)", "")
+            tools.replace_in_file(cmakelists, "if (ZLIB_FOUND)", "if(0)")
+        if not self.options.with_bzip2:
+            tools.replace_in_file(cmakelists, "find_package(BZip2)", "")
+            tools.replace_in_file(cmakelists, "if (BZIP2_FOUND)", "if(0)")
+        if self._has_with_brotli_option:
+            # the custom FindBrotliDec of upstream is too fragile
+            tools.replace_in_file(cmakelists,
+                                  "find_package(BrotliDec REQUIRED)",
+                                  "find_package(Brotli REQUIRED)\n"
+                                  "set(BROTLIDEC_FOUND 1)\n"
+                                  "set(BROTLIDEC_LIBRARIES \"Brotli::Brotli\")")
+            if not self.options.with_brotli:
+                tools.replace_in_file(cmakelists, "find_package(BrotliDec)", "")
+                tools.replace_in_file(cmakelists, "if (BROTLIDEC_FOUND)", "if(0)")
```

Comment:
This is not 100% correct, if I'm not mistaken.

You will first unconditionally change `find_package(BrotliDec REQUIRED)` to `find_package(Brotli REQUIRED)` + some extra sets.

But then under `if not self.options.with_brotli` you replace `find_package(BrotliDec)` with empty string. But it won't be in the file, because you already replaced it several lines above.

Reply:
it's not the same string, you can test locally if you want with and without brotli.

Reply:
Oh, I see now, the strings are different 👍